src/SpannerServiceProvider.php

```diff
@@ -71,22 +77,22 @@ protected function parseConfig(array $config, $name)
     }
 
     /**
-     * @param array $sessionPoolConfig
+     * @param array<string, mixed> $sessionPoolConfig
      * @return SessionPoolInterface
      */
     protected function createSessionPool(array $sessionPoolConfig): SessionPoolInterface
     {
-        $cachePath = storage_path(implode(DIRECTORY_SEPARATOR, ['framework', 'cache', 'spanner']));
-        return new CacheSessionPool(new FileCacheAdapter('session', $cachePath), $sessionPoolConfig);
+        $cachePath = storage_path(implode(DIRECTORY_SEPARATOR, ['framework', 'spanner']));
+        $adapter = new FilesystemAdapter('session', 0, $cachePath);
```

Comment:
```suggestion
        $adapter = new FilesystemAdapter(namespace: 'session', directory: $cachePath);
```

How about using `Named Arguments`?

Reply:
I don't understand the need for it?

Reply:
The `$defaultLifetime` (the second argument) is the same as the default value, but the intent is to explicitly leave it to the default value since it is not of interest in this use case.

Reply:
Actually, unlimited lifetime is what is expected here, so explicitly defining it would be a good thing here though?

Reply:
Okay, if it is an intentional designation, no problem.
I thought I was forced to specify the default value of `$defaultLifetime` in order to specify `$directory` (the third argument), but I was mistaken.

Reply:
Apologies for the confusion.

Before this fix, the FileCacheAdapter we used did not allow you to specify the lifetime and was expected to be unlimited (although internally, it did not specify so it was using the default behavior).

In this case, either solution works, because the default fall back is unlimited anyway, but I figured explicit would be best here.